app/views/users/profile/_summary.html.erb

```diff
@@ -39,7 +39,7 @@
         </li>
     <% end %>
 
-    <% if presenter.display_hire_me && current_user != presenter.user %>
+    <% if presenter.display_hire_me && !presenter.user_same_as?(current_user) %>
```

Comment:
could the  `!presenter.user_same_as?(current_user)` be pulled into the presenter.display_hire_me method?  no need to expose the logic in the view right?  I guess maybe that could be another ticket and PR but then again ... up to you :-)